manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,639 @@
+     *
+     * @param command string of the command, without preformatting.
+     * List of valid commands can be found in Teltonika's website.
+     * @param device A {@link TeltonikaDevice} that is currently subscribed, to which to send the message to.
+     */
+    private void sendCommandToTeltonikaDevice(String command, TeltonikaDevice device) {
+        HashMap<String, String> cmdtest = new HashMap<>();
+        cmdtest.put("CMD", command);
+
+        mqttBrokerService.publishMessage(device.commandTopic, cmdtest, MqttQoS.EXACTLY_ONCE);
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return LOG;
+    }
+    @Override
+    public boolean checkCanSubscribe(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        // Skip standard checks
+        if (!canSubscribe(connection, securityContext, topic)) {
+            getLogger().fine("Cannot subscribe to this topic, topic=" + topic + ", connection" + connection);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the Subscribing client should be allowed to subscribe to the topic that is handled by this Handler.
+     * For Teltonika device endpoints, we need the fourth token (Index 3) to be a valid IMEI number.
+     * We do that by checking using IMEIValidator.
+     */
+    // To be removed when auto-provisioning works
+    @Override
+    public boolean canSubscribe(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        if(topic.getTokens().size() < 5){
+            getLogger().warning(MessageFormat.format("Topic {0} is not a valid Topic. Please use a valid Topic.", topic.getString()));
+            return false;
+        }
+        long imeiValue = 0;
+        try{
+            imeiValue = Long.parseLong(topic.getTokens().get(3));
+        }catch (NumberFormatException e){
+            getLogger().warning(MessageFormat.format("IMEI {0} is not a valid IMEI value. Please use a valid IMEI value.", topic.getTokens().get(3)));
+            return false;
+        }
+                // IMEI should already be checked
+        return  Objects.equals(topic.getTokens().get(2), TELTONIKA_DEVICE_TOKEN) &&
+                IMEIValidator.isValidIMEI(imeiValue) &&
+                (
+                        Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_RECEIVE_TOPIC) ||
+                        Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_SEND_TOPIC)
+                );
+    }
+
+    /**
+     * Overrides MQTTHandler.checkCanPublish for this specific Handler,
+     * until secure Authentication and Auto-provisioning
+     * of Teltonika Devices is created.
+     * To be removed after implementation is complete.
+     */
+    @Override
+    public boolean checkCanPublish(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        return true;
```

Comment:
The reason I'm currently overriding this is because of avoiding Keycloak authentication, something currently not possible for Teltonika devices. This will be removed when authentication for the devices is properly implemented